python/test/data_extraction/test_export_data.py

```diff
@@ -65,4 +66,16 @@ def test_export_untagged_content(self):
                                  side_effect=self.return_input_or_output(io.StringIO(input_string), output)):
             export_data.export_untagged_content()
             expected = [{"title":"title1"}]
-            self.assertListEqual(expected, json.loads(output.buffer)["items"])
\ No newline at end of file
+            self.assertListEqual(expected, json.loads(output.buffer)["items"])
+
+    @freeze_time("2018-03-07")
+    def test_export_metadata(self):
+        output = MockIO()
+        input_string = json.dumps([content_with_taxons, content_without_taxons])
+
+        with unittest.mock.patch('gzip.open',
+                                 side_effect=self.return_input_or_output(io.StringIO(input_string), output)):
+            export_data.export_untagged_content()
+            expected = {"date": "2018-03-07 00:00:00", 
+                        "code": Repo(search_parent_directories=True).head.commit.hexsha}
```

Comment:
Can Repo be mocked?

Reply:
I looked and couldn't see a way - more symptomatic of the script having internal dependencies, which could be passed-in. Think that's going a bit far, though.

Reply:
Could 'patch' Repo - something similar is happening in this file when patching gzip.open

Reply:
I think that only applies to a single method invocation. The Repo bit is actually a constructor - I don't think you can override those.

Reply:
You could use:
```
    @patch('data_extraction.export_data.Repo')
    def test_export_metadata(self, MockedRepo):
        MockedRepo.return_value = Mock(commit=Mock(hexsha='myhexsha'))
```

Reply:
That works - thanks Jos!